File: 0283-move-zeroes/solution.go
func moveZeroes(nums []int)  {

    // find first zero idx
    zeroIdx := -1
    for i, v := range nums {
        if v == 0 {
            zeroIdx = i
            break
        }
    }

    if zeroIdx == -1 {
        return
    }

    ptr := zeroIdx + 1
    for ; ptr < len(nums); ptr++ {
        if nums[ptr] != 0 {

            temp := nums[ptr]
            nums[ptr] = nums[zeroIdx]
            nums[zeroIdx] = temp

            zeroIdx++
        }
    }
}

func swap(i, j int, arr []int) {
    
}

File: 0283-move-zeroes/solution.py
class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        ptr_write = 0
        n = len(nums)
        count_zero = 0
        for i in range(n):
            if nums[i] == 0:
                count_zero += 1
            else:
                nums[ptr_write] = nums[i]
                ptr_write += 1
        
        for i in range(n - 1, n - 1 - count_zero, -1):
            nums[i] = 0
        
